src/api/public/notes/notes.controller.ts

```diff
@@ -242,6 +246,177 @@ export class NotesController {
     );
   }
 
+  @UseInterceptors(GetNoteInterceptor)
+  @Permissions(Permission.READ)
+  @UseGuards(TokenAuthGuard, PermissionsGuard)
+  @Get(':noteIdOrAlias/metadata/permissions')
+  @ApiOkResponse({
+    description: 'Get the permissions for a note',
+    type: NotePermissionsDto,
+  })
+  @FullApi
+  async getPermissions(
+    @RequestUser() user: User,
+    @RequestNote() note: Note,
+  ): Promise<NotePermissionsDto> {
+    return await this.noteService.toNotePermissionsDto(note);
+  }
+
+  @UseInterceptors(GetNoteInterceptor)
+  @Permissions(Permission.OWNER)
+  @UseGuards(TokenAuthGuard, PermissionsGuard)
+  @Get(':noteIdOrAlias/metadata/permissions/users/:userName')
```

Comment:
Probably you meant this?

```suggestion
  @Put(':noteIdOrAlias/metadata/permissions/users/:userName')
```

Reply:
Yes, copy paste mistake. Fixed.